server/internal/core/file/file_test.go

```diff
@@ -0,0 +1,54 @@
+package file
+
+import (
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/oxidrive/oxidrive/server/internal/core/user"
+	"github.com/oxidrive/oxidrive/server/internal/testutil"
+)
+
+func Test_Create(t *testing.T) {
+	t.Run("creates a new valid file", func(t *testing.T) {
+		t.Parallel()
+
+		file, err := Create(nil, "this/is/a/directory/filename.txt", 5, user.ID(testutil.Must(uuid.NewV7())))
+
+		assert.NotNil(t, file)
+		assert.NoError(t, err)
+	})
+
+	t.Run("returns an error with an invalid path", func(t *testing.T) {
+		t.Parallel()
+
+		file, err := Create(nil, "/this/is/a/directory/filename.txt", 5, user.ID(testutil.Must(uuid.NewV7())))
```

Comment:
Is this really invalid? I know technically it's an absolute path, but it could simply be joined with our mountpoint and prefix right?

Reply:
If we decide to accept absolute paths, there are two consequences:
- `/this/is/a/directory/filename.txt` will be equal to `this/is/a/directory/filename.txt`
- It adds a layer of complexity in the validation, we have to check for this `/this/../../../../filename.txt`, which is not necessary complex

It could be a good idea, right now with the API you cannot upload a file using an absolute path, unless you modify the header of the form-data
```sh
curl -X POST -H "Content-Type: multipart/form-data" -H "Authorization: Bearer <TOKEN>" -F file=@/root/CODEOWNERS http://127.0.0.1:8888/api/files
```

WDYT?

Reply:
Maybe we could simply normalize the path by adding an initial `/` if it's not already there?